app/controllers/DeleteAllFixedEstablishmentController.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.data.Validated.Valid
+import connectors.RegistrationConnector
+import controllers.actions._
+import controllers.amend.{routes => amendRoutes}
+import logging.Logging
+import models.{AmendMode, Country, Index, UserAnswers}
+import models.euDetails.{EuConsumerSalesMethod, EuDetails}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import queries.{AllEuDetailsQuery, EuDetailsQuery}
+import repositories.AuthenticatedUserAnswersRepository
+import services.{RegistrationService, RegistrationValidationService}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.FutureSyntax.FutureOps
+import views.html.DeleteAllFixedEstablishmentView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class DeleteAllFixedEstablishmentController @Inject()(
+                                                       override val messagesApi: MessagesApi,
+                                                       cc: AuthenticatedControllerComponents,
+                                                       registrationConnector: RegistrationConnector,
+                                                       registrationService: RegistrationService,
+                                                       registrationValidationService: RegistrationValidationService,
+                                                       view: DeleteAllFixedEstablishmentView,
+                                                       authenticatedUserAnswersRepository: AuthenticatedUserAnswersRepository
+                                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with Logging {
+
+  protected val controllerComponents: MessagesControllerComponents = cc
+
+  def onPageLoad: Action[AnyContent] = cc.authAndGetOptionalData(Some(AmendMode)).async {
+    implicit request =>
+      (for {
+        maybeRegistration <- registrationConnector.getRegistration()
```

Comment:
you shouldn't have to do this, you can use request.registration which should be a Option[Registration], saves having to call the connector again, when it should already be in scope